src/backend/base/langflow/components/vectorstores/Weaviate.py

```diff
@@ -37,6 +42,13 @@ class WeaviateVectorStoreComponent(LCVectorStoreComponent):
         ),
         BoolInput(name="search_by_text", display_name="Search By Text", advanced=True),
     ]
+    index_name: str
+
+    def get_valid_index_name(self, client: weaviate.Client, index_name: str) -> str:
```

Comment:
Based on https://forum.weaviate.io/t/feature-or-bug-first-letter-of-class-name-gets-capitalized/1203, I'm debating on whether this is a feature we need to support. It sounds like Weaviate will auto-capitalize the index name for us under the covers, which, while not ideal, will not cause any errors to be thrown here, correct? 

They also mention that this won't happen when using the v4 python client. (https://weaviate.io/developers/weaviate/client-libraries/python/v3_v4_migration) - I haven't looked into the impl details or exact behavior though.

Reply:
@edwinjosechittilappilly Remind me where we ended up with this - are we blocked on updating to v4?